evalml/tests/automl_tests/test_automl.py

```diff
@@ -412,6 +412,23 @@ def test_automl_bad_data_check_parameter_type():
         automl.search(data_checks=[MockDataCheckErrorAndWarning])
 
 
+class MockDataCheckObjective(DataCheck):
+    def __init__(self, objective):
+        self.objective_name = get_objective(objective).name
+
+    def validate(self, X, y):
+        return {"warnings": [], "errors": []}
+
+
+@pytest.mark.parametrize("data_checks", [DataChecks([MockDataCheckObjective],
+                                                    data_check_params={"MockDataCheckObjective": {"objective": 'R2'}})])
+def test_automl_passes_correct_objective_name_to_data_check(data_checks, X_y_regression):
+    X, y = X_y_regression
+    automl = AutoMLSearch(X, y, problem_type=ProblemTypes.REGRESSION)
+    automl.search(data_checks=data_checks)
+    assert automl._validate_data_checks(data_checks).data_checks[0].objective_name == MockDataCheckObjective("R2").objective_name
```

Comment:
Can we also test that `DefaultDataChecks` is initialized with the right objective name if `data_checks='auto'`?